src/bootstrap/src/ferrocene/run.rs

```diff
@@ -111,3 +115,159 @@ impl Step for TraceabilityMatrix {
         html_output
     }
 }
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
+pub(crate) struct CertifiedCoreSymbols {
+    pub(super) build_compiler: Compiler,
+    pub(super) target: TargetSelection,
+}
+
+impl Step for CertifiedCoreSymbols {
+    type Output = PathBuf;
+    const DEFAULT: bool = false;
+    const IS_HOST: bool = true;
+
+    fn should_run(run: ShouldRun<'_>) -> ShouldRun<'_> {
+        run.path(SYMBOL_PATH).alias("certified-core-symbols")
+    }
+
+    fn make_run(run: RunConfig<'_>) {
+        let build_compiler = run.builder.compiler(run.builder.top_stage.max(1), run.build_triple());
+        run.builder.ensure(CertifiedCoreSymbols { build_compiler, target: run.target });
+    }
+
+    fn run(self, builder: &Builder<'_>) -> Self::Output {
+        let CertifiedCoreSymbols { build_compiler, target } = self;
+        let symbol_report = builder.ensure(SymbolReport { target_compiler: build_compiler });
+
+        let certified_target = self
+            .target
+            .certified_equivalent()
+            .expect(&format!("no certified equivalent exists for target \"{target}\""));
+
+        // c.f. check::std
+        let mut cargo = Cargo::new(
+            builder,
+            build_compiler,
+            Mode::Std,
+            SourceType::InTree,
+            certified_target,
+            Kind::Check,
+        );
+        let crates = vec!["core".to_owned()]; // currently, only core is certified
+        for krate in &*crates {
+            cargo.arg("-p").arg(krate);
```

Comment:
Isn't this already done in `std_cargo` now?

Reply:
Yes, good catch.